.github/workflows/ghcr.yml

```diff
@@ -141,28 +96,54 @@ jobs:
       - name: Upload Docker image as artifact
         uses: actions/upload-artifact@v4
         with:
-          name: ${{ matrix.image }}-docker-image-${{ matrix.platform }}
-          path: /tmp/${{ matrix.image }}_image_${{ matrix.platform }}.tar
+          name: ${{ matrix.image }}_${{ steps.build.outputs.last_tag }}_${{ matrix.platform }}
+          path: /tmp/${{ matrix.image }}_${{ steps.build.outputs.last_tag }}_${{ matrix.platform }}.tar
           retention-days: 14
+      - name: Capture last tag
+        id: capture-last-tag
+        run: |
+          last_tag=$(cat tags.txt | awk '{print $NF}')
+          echo "${{ matrix.image }}_${{ matrix.base_image }}_${{ matrix.platform }}=$last_tag" > last_tag.txt
+      - name: Upload last tag as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: last-tag-${{ matrix.image }}-${{ steps.build.outputs.last_tag }}-${{ matrix.platform }}
+          path: /tmp/last-tag.txt
+          retention-days: 1
+
+  prepare_test_matrix:
+    needs: ghcr_build_runtime
+    runs-on: ubuntu-latest
+    outputs:
+      test_matrix: ${{ steps.set-matrix.outputs.test_matrix }}
+    steps:
+      - name: Download last tags
+        uses: actions/download-artifact@v4
+        with:
+          pattern: last-tag-*
+          merge-multiple: true
+      - name: Set up test matrix
+        id: set-matrix
+        run: |
+          matrix=$(cat /tmp/last-tag-* | jq -R -s -c 'split("\n")[:-1] | map(select(length > 0)) | map(split("=")) | map({"image": .[0] | split("_")[0], "base_image": .[0] | split("_")[1], "platform": .[0] | split("_")[2], "last_tag": .[1]})')
```

Comment:
This line scares me! 😂

Reply:
This is from sonnet - let's see if the results make sense or not 🔮